etl/churn.kp/knowledge.md

```diff
@@ -511,19 +536,39 @@ def compute_churn_week(df, week_start, bucket, prefix):
             ),(
                 1,  # active users 
                 x.get('usage_hours', 0),
-                x.get('squared_usage_hours',0) 
+                x.get('squared_usage_hours',0),
+                x.get('total_uri_count', 0),
+                x.get('unique_domains_count', 0)
             )
         )
     )
 
     def reduce_func(x, y):
-        return (x[0] + y[0], # Sum active users
-                x[1] + y[1], # Sum usage_hours
-                x[2] + y[2]) # Sum squared_usage_hours
-
+        return tuple(map(sum, zip(x, y)))
+    
     aggregated = countable.reduceByKey(reduce_func)
 
     records_df = aggregated.map(lambda x: x[0] + x[1]).toDF(record_columns)
+    
+    # Apply some post-processing for other aggregates (i.e. unique_domains_count). This
+    # needs to be done when you want something other than just a simple sum
+    def average(total, n):
+        if not n:
+            return 0.0
+        return float(total)/n
+    average_udf = F.udf(average, DoubleType())
+    
+    # Create new derived columns and any unecessary ones
```

Comment:
nit: s/Create new derived columns and any unecessary ones/Create new derived columns and drop any unnecessary ones/